Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```
@@ -207,20 +242,24 @@
         return;
     }
 
-    auto maybeSidebar = getOrCreateSidebar(*window);
+    if (!hasActiveUserGesture(*tab)) {
+        completionHandler(toWebExtensionError(nil, nil, @"close can only be called in response to a user action"));
+        return;
+    }
+
+    auto maybeSidebar = getOrCreateSidebar(*tab);
     if (!maybeSidebar) {
-        completionHandler(toWebExtensionError(nil, nil, @"the sidebar could not be created"));
+        completionHandler(toWebExtensionError(nil, nil, @"could not get sidebar for active tab"));
         return;
     }
 
-    auto& sidebar = maybeSidebar.value();
-    if (!sidebar->canProgrammaticallyCloseSidebar()) {
+    Ref sidebar = WTFMove(maybeSidebar.value());
+    if (!canProgrammaticallyCloseSidebar()) {
         completionHandler(toWebExtensionError(nil, nil, @"it is not implemented"));
         return;
     }
```

Should this be earlier, so it fails faster?

---

Also add API name for first parameter.